lib/matplotlib/backends/backend_webagg_core.py

```diff
@@ -227,17 +230,17 @@ def get_diff_image(self):
         if self._png_is_old:
             renderer = self.get_renderer()
 
+            pixels = np.asarray(renderer.buffer_rgba())
             # The buffer is created as type uint32 so that entire
             # pixels can be compared in one numpy call, rather than
             # needing to compare each plane separately.
-            buff = (np.frombuffer(renderer.buffer_rgba(), dtype=np.uint32)
-                    .reshape((renderer.height, renderer.width)))
-
-            # If any pixels have transparency, we need to force a full
-            # draw as we cannot overlay new on top of old.
-            pixels = buff.view(dtype=np.uint8).reshape(buff.shape + (4,))
+            buff = pixels.view(np.uint32).squeeze(2)
 
-            if self._force_full or np.any(pixels[:, :, 3] != 255):
+            if (self._force_full
+                    or buff.shape != self._last_buff.shape
```

Comment:
may be nice to add a comment as well about this line. It took me a while to hunt down why it was ok to compare the initial `np.empty((0,0))` with the buffer and this line is the escape